src/Simplex/Chat/Messages.hs

```diff
@@ -150,13 +151,21 @@ data ChatItem (c :: ChatType) (d :: MsgDirection) = ChatItem
   { chatDir :: CIDirection c d,
     meta :: CIMeta c d,
     content :: CIContent d,
+    mentions :: [MentionedMember], -- to prevent loading all members from UI
     formattedText :: Maybe MarkdownList,
     quotedItem :: Maybe (CIQuote c),
     reactions :: [CIReactionCount],
     file :: Maybe (CIFile d)
   }
   deriving (Show)
 
+data MentionedMember = MentionedMember
+  { memberViewName :: Text, -- display name or alias
+    memberRole :: GroupMemberRole
```

Comment:
why role is needed here? also why memberId is not in this type and only in markdown?

Reply:
to show it as **ep** (*admin*)

Reply:
memberId should be included, I initially made it a Map.